compiler/java/src/main/java/com/c/lexer/Lexer.java

```diff
@@ -1,29 +1,635 @@
+        return !commentBlockCloseIdentifier.isEmpty() && reader.lookAheads().startsWith("}" + commentBlockCloseIdentifier);
+    }
+
+
+    // ------------------------------------------ Consume --------------------------------------------------------------
+
+    /**
+     * Consumes EOF (end of file).
+     *
+     * @return a Token representing EOF
+     */
+    private Token consumeEOF() {
+        return constructToken(TokenType.EOF);
+    }
+
+    /**
+     * Consumes a line comment identifier (e.g. in Java "//").
+     * This will switch to the SECTION context.
+     *
+     * @return a Token representing that line comment identifier
+     * @throws UnexpectedCharException if a char - read by the FileReader - was not the expected one
+     */
+    private Token consumeCommentLineIdentifier() throws UnexpectedCharException {
+        for (char c : commentLineIdentifier.toCharArray()) {
+            reader.expect(c);
+        }
+        currentContext = Context.SECTION;
+        return constructToken(TokenType.COMMENT_LINE_IDENTIFIER);
+    }
+
+    /**
+     * Consumes an open block comment identifier (e.g. in Java "/*").
+     * This will switch to the SECTION context.
+     *
+     * @return a Token representing that open block comment identifier
+     * @throws UnexpectedCharException if a char - read by the FileReader - was not the expected one
+     */
+    private Token consumeCommentBlockOpenIdentifier() throws UnexpectedCharException {
+        for (char c : commentBlockOpenIdentifier.toCharArray()) {
+            reader.expect(c);
+        }
+        currentContext = Context.SECTION;
+        return constructToken(TokenType.COMMENT_BLOCK_OPEN_IDENTIFIER);
+    }
+
+    /**
+     * Consumes a close block comment identifier (e.g. in Java "(star)/" where (star) is equal to *).
+     * This will switch to the SECTION context.
+     *
+     * @return a Token representing that close block comment identifier
+     * @throws UnexpectedCharException if a char - read by the FileReader - was not the expected one
+     */
+    private Token consumeCommentBlockCloseIdentifier() throws UnexpectedCharException {
+        for (char c : commentBlockCloseIdentifier.toCharArray()) {
+            reader.expect(c);
+        }
+        currentContext = Context.ARBITRARY;
+        return constructToken(TokenType.COMMENT_BLOCK_CLOSE_IDENTIFIER);
+    }
+
+    /**
+     * Consumes a payload comment identifier (same identifier as line comments).
+     * This will switch to the SECTION context.
+     *
+     * @return a Token representing that payload comment identifier
+     * @throws UnexpectedCharException if a char - read by the FileReader - was not the expected one
+     */
+    private Token consumeCommentPayloadIdentifier() throws UnexpectedCharException {
+        for (char c : commentPayloadIdentifier.toCharArray()) {
+            reader.expect(c);
+        }
+        currentContext = Context.SECTION;
```

Comment:
So true... This is a good sign that our code is pretty flexible since all the tests passed nevertheless ;) I was lucky since we always stay in the while loop within `consumePayload()`, so this context change didn't affect us. Got rid of it now.